File: WagsMediaGenerator/Models/Podcast.cs
namespace WagsMediaGenerator.Models;

public class Podcast
{
    public int PodcastId { get; set; }
    
    public int PodcastCategoryId { get; set; }

    public string Name { get; set; } = string.Empty;
    
    public string Link { get; set; } = string.Empty;
    
    public string CoverImageUrl { get; set; } = string.Empty;

    public Tag Category { get; set; } = new();
}

File: api/src/models/podcast.ts
export type PodcastCategory = {
    podcastCategoryId: number;
    name: string;
    colorCode: string;
    podcastCount?: number;
}

export type Podcast = {
    podcastId: number;
    podcastCategoryId: number;
    name: string;
    link: string;
    coverImageUrl: string;
    category: PodcastCategory;
}

export type PodcastQueryReturn = {
    PodcastId: number;
    Name: string;
    Link: string;
    CoverImageUrl: string;
    PodcastCategoryId: number;
    PodcastCategoryName: string;
    ColorCode: string;
}

export type PodcastCategoryQueryReturn = {
    PodcastCategoryId: number;
    Name: string;
    ColorCode: string;
    PodcastCount: number;
}